### PR TITLE
[location] update docs for default LocationAccuracy

### PR DESCRIPTION
**Pull request by johnculviner**
# Why
The defaults for `LocationAccuracy` are not documented and are potentially too granular for some use-cases potentially leading to excess time for location resolution or [even errors](https://github.com/expo/expo/issues/5487).

This helps surface the defaults which will allow people to more readily change them if they differ from their application's needs.

Defaults determined by below code:
- [Android](https://github.com/expo/expo/blob/main/packages/expo-location/android/src/main/java/expo/modules/location/LocationHelpers.kt#L167)
- [iOS](https://github.com/expo/expo/blob/main/packages/expo-location/ios/EXLocation.m#L41)

# How
I followed the guides here:
- https://github.com/expo/expo/blob/main/CONTRIBUTING.md
- https://github.com/expo/expo/blob/main/docs/README.md

# Test Plan
- I followed the guide in docs on regenerating docs based on comments and completed the - https://github.com/expo/expo/blob/main/CONTRIBUTING.md

# Checklist
- [x] Documentation is up to date to reflect these changes (eg: https://docs.expo.dev and README.md).
- [x] Conforms with the [Documentation Writing Style Guide](https://github.com/expo/expo/blob/main/guides/Expo%20Documentation%20Writing%20Style%20Guide.md)
- [x] This diff will work correctly for `npx expo prebuild` & EAS Build (eg: updated a module plugin).

### CODE REVIEW

File: packages/expo-location/src/Location.types.ts [hunk 1]
```diff
@@ -14,7 +14,7 @@ export enum LocationAccuracy {
    */
   Low = 2,
   /**
-   * Accurate to within one hundred meters.
+   * Accurate to within one hundred meters. (Default)
```
Comment (lukmccall) [inline packages/expo-location/src/Location.types.ts:17]:
```suggestion
   * Accurate to within one hundred meters.
```

File: packages/expo-location/src/Location.types.ts [hunk 2]
```diff
@@ -105,7 +105,8 @@ export type LocationOptions = {
   /**
    * Location manager accuracy. Pass one of `Accuracy` enum values.
    * For low-accuracies the implementation can avoid geolocation providers
-   * that consume a significant amount of power (such as GPS).
+   * that consume a significant amount of power (such as GPS). Default is
+   * LocationAccuracy.Balanced
```
Comment (lukmccall) [inline packages/expo-location/src/Location.types.ts:109]:
```suggestion
   * that consume a significant amount of power (such as GPS).
   *
   * @default LocationAccuracy.Balanced
```
We have a special annotation that specifies the default value.

**Comment by johnculviner** [comment]
Great, thanks for the suggestions. I updated the code with them and regenerated the docs with `./bin/et generate-docs-api-data --packageName expo-location` with the changes

**Comment by johnculviner** [comment]
I regenerated the code built code per the suggested changes

**Review by lukmccall — APPROVED** [review]
Thanks! Looking good

**Pull request merged**